lib/ansible/module_utils/network/aci/msc.py

```diff
@@ -0,0 +1,299 @@
+        self.module = module
+        self.params = module.params
+        self.result = dict(changed=False)
+        self.headers = {'Content-Type': 'text/json'}
+
+        # normal output
+        self.existing = dict()
+
+        # info output
+        self.previous = dict()
+        self.proposed = dict()
+        self.sent = dict()
+
+        # debug output
+        self.filter_string = ''
+        self.method = None
+        self.path = None
+        self.response = None
+        self.status = None
+        self.url = None
+
+        # Ensure protocol is set
+        self.params['protocol'] = 'https' if self.params.get('use_ssl', True) else 'http'
+
+        # Set base_uri
+        if 'port' in self.params and self.params['port'] is not None:
+            self.baseuri = '{protocol}://{host}:{port}/api/v1/'.format(**self.params)
+        else:
+            self.baseuri = '{protocol}://{host}/api/v1/'.format(**self.params)
+
+        if self.module._debug:
+            self.module.warn('Enable debug output because ANSIBLE_DEBUG was set.')
+            self.params['output_level'] = 'debug'
+
+        if self.params['password']:
+            # Perform password-based authentication, log on using password
+            self.login()
+        else:
+            self.module.fail_json(msg="Parameter 'password' is required for authentication")
+
+    def login(self):
+        ''' Log in to MSC '''
+
+        # Perform login request
+        self.url = urljoin(self.baseuri, 'auth/login')
+        payload = {'username': self.params['username'], 'password': self.params['password']}
+        resp, auth = fetch_url(self.module,
+                               self.url,
+                               data=json.dumps(payload),
+                               method='POST',
+                               headers=self.headers,
+                               timeout=self.params['timeout'],
+                               use_proxy=self.params['use_proxy'])
+
+        # Handle MSC response
+        if auth['status'] != 201:
+            self.response = auth['msg']
+            self.status = auth['status']
+            self.fail_json(msg='Authentication failed: {msg}'.format(**auth))
+
+        payload = json.loads(resp.read())
+
+        self.headers['Authorization'] = 'Bearer {token}'.format(**payload)
+
+    def request(self, path, method=None, data=None):
+        ''' Generic HTTP method for Meraki requests. '''
+        self.path = path
+
+        if method is not None:
+            self.method = method
+
+        self.url = urljoin(self.baseuri, path)
+        resp, info = fetch_url(self.module,
+                               self.url,
+                               headers=self.headers,
+                               data=json.dumps(data),
+                               method=self.method,
+                               timeout=self.params['timeout'],
+                               use_proxy=self.params['use_proxy'],
```

Comment:
I think it's likely the proxy code won't work as I haven't been able to test it well under Meraki

Reply:
It works, at least it did for ACI because fetch_url does this for us.

Reply:
If your system has `http_proxy` or `https_proxy` environment variables declared, fetch_url picks them up automatically. The `use_proxy` parameter would disable this behavior (which is enabled by default).